test_communication/CMakeLists.txt

```diff
@@ -123,7 +123,8 @@ if(BUILD_TESTING)
         INPUT "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber_cpp${test_suffix}.py.configured"
       )
 
-      ament_add_nose_test(publisher_subscriber_cpp${test_suffix} "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber_cpp${test_suffix}_$<CONFIGURATION>.py" TIMEOUT 15)
+      ament_add_nose_test(publisher_subscriber_cpp${test_suffix} "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber_cpp${test_suffix}_$<CONFIGURATION>.py" TIMEOUT 15
+        ENV RCL_ASSERT_RMW_ID_MATCHES=${rmw_implementation})
```

Comment:
I guess the `rmw_implementation` needs to be passed down into the `.py` file and then the env be set by `launch`.

Can we just roll that change into the pr that re-enables all the tests that were previously disabled?